src/php/iterators/php_htable_iterator.h

```diff
@@ -10,6 +10,7 @@ typedef struct ds_htable_iterator {
     uint32_t                 position;
     ds_htable_bucket_t      *bucket;
     ds_htable_t             *table;
+    zval                    *obj;
```

Comment:
This should be a `zval` into which you ZVAL_COPY, or a `zend_object*` (in with case you can use GC_ADDREF and OBJ_RELEASE as the refcount operations). You can't keep just a pointer here, as the lifetime of the iterator may exceed the lifetime of that zval.